upickle/implicits/src-3/upickle/implicits/macros.scala

```diff
@@ -199,6 +199,54 @@ def tagKeyImpl[T](using Quotes, Type[T])(thisOuter: Expr[upickle.core.Types with
     case None => '{${thisOuter}.tagName}
   }
 
+inline def applyConstructor[T](params: Array[Any]): T = ${ applyConstructorImpl[T]('params) }
+def applyConstructorImpl[T](using quotes: Quotes, t0: Type[T])(params: Expr[Array[Any]]): Expr[T] =
+  import quotes.reflect._
+  def apply(typeApply: Option[List[TypeRepr]]) = {
+    val tpe = TypeRepr.of[T]
+    val companion: Symbol = tpe.classSymbol.get.companionModule
+    val constructorSym = tpe.typeSymbol.primaryConstructor
+    val constructorParamSymss = constructorSym.paramSymss
```

Comment:
Are you sure that those always exist? Quotes API has a bad habit of returning `Symbol.noSymbol` instead of `None`. It is not obvious from the type signature that those methods can fail.